asm/instruction.go

```diff
@@ -40,10 +39,12 @@ func (ins *Instruction) Unmarshal(r io.Reader, bo binary.ByteOrder) (uint64, err
 	}
 
 	ins.OpCode = bi.OpCode
-	ins.Dst = bi.Registers.Dst()
-	ins.Src = bi.Registers.Src()
 	ins.Offset = bi.Offset
 	ins.Constant = int64(bi.Constant)
+	ins.Dst, ins.Src, err = bi.Registers.Unmarshal(bo)
+	if err != nil {
+		return 0, xerrors.Errorf("unmarshal registers: %s", err)
```

Comment:
Wrap the `error` value with `%w`. Also maybe adjust the error message a bit:

```suggestion
		return 0, xerrors.Errorf("can't unmarshal registers: %w", err)
```